packages/components/src/toolbar-item/index.native.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { View as BaseToolbarItem } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useContext } from '@wordpress/element';
+import warning from '@wordpress/warning';
+
+/**
+ * Internal dependencies
+ */
+import ToolbarContext from '../toolbar-context';
+
+function ToolbarItem( { children, ...props }, ref ) {
+	const accessibleToolbarState = useContext( ToolbarContext );
+
+	if ( typeof children !== 'function' ) {
+		warning(
+			'`ToolbarItem` is a generic headless component that accepts only function children props'
+		);
+		return null;
+	}
+
+	const allProps = { ...props, ref, 'data-experimental-toolbar-item': true };
+
+	if ( ! accessibleToolbarState ) {
+		return children( allProps );
+	}
+
+	return (
+		<BaseToolbarItem { ...accessibleToolbarState } { ...allProps }>
+			{ ( htmlProps ) =>
+				// Overwriting BaseToolbarItem's onMouseDown since it disables drag
+				// and drop
+				children( { ...htmlProps, onMouseDown: allProps.onMouseDown } )
+			}
+		</BaseToolbarItem>
+	);
```

Comment:
Does this work on RN?
```jsx
<View prop="value">{ ( props ) => children( props ) }</View>
```
I would say that this is not needed at all. And the code would never reach to that point any way because `accessibleToolbarState` should be always falsy on RN. So `return children( { ...props, ref } )` is probably all you need.

Reply:
It works, but feel free to push to the PR, with your changes, and I can test it.